src/ray/core_worker/store_provider/plasma_store_provider.cc

```diff
@@ -178,9 +193,14 @@ Status CoreWorkerPlasmaStoreProvider::Get(
     }
 
     size_t previous_size = remaining.size();
-    RAY_RETURN_NOT_OK(FetchAndGetFromPlasmaStore(remaining, batch_ids, batch_timeout,
-                                                 /*fetch_only=*/false, task_id, results,
-                                                 got_exception));
+    // This is a separate IPC from the FetchAndGet in direct call mode.
+    if (ctx.CurrentTaskIsDirectCall() && ctx.ShouldReleaseResourcesOnBlockingCalls()) {
+      RAY_RETURN_NOT_OK(raylet_client_->NotifyDirectCallTaskBlocked());
```

Comment:
We could use the in_direct_call flag to directly call `HandleDirectCallTaskBlocked` from the node_manager, in the `ProcessFetchOrReconstructMessage`.

Reply:
The reason that doesn't work is that we need to check if we are also in the main thread and not in an actor method. We could pass this metadata all the way to the raylet, but it is simpler to decide here and send the IPC separately.